src/Network/Freddy.hs

```diff
@@ -121,39 +126,42 @@ deliver connection request = do
     AMQP.msgExpiration   = Request.expirationInMs request
   }
 
-  AMQP.publishMsg (amqpChannel connection) "" (Request.queueName request) msg
-  AMQP.publishMsg (amqpChannel connection) topicExchange (Request.queueName request) msg
+  AMQP.publishMsg (amqpProduceChannel connection) "" (Request.queueName request) msg
+  AMQP.publishMsg (amqpProduceChannel connection) topicExchange (Request.queueName request) msg
 
 respondTo :: Connection -> QueueName -> (Delivery -> IO ()) -> IO Consumer
 respondTo connection queueName callback = do
-  let channel = amqpChannel connection
-  AMQP.declareQueue channel AMQP.newQueue {AMQP.queueName = queueName}
-  tag <- AMQP.consumeMsgs channel queueName AMQP.NoAck (replyCallback callback channel)
-  return Consumer { consumerChannel = channel, consumerTag = tag }
+  let produceChannel = amqpProduceChannel connection
+  consumeChannel <- AMQP.openChannel . amqpConnection $ connection
+  AMQP.declareQueue consumeChannel AMQP.newQueue {AMQP.queueName = queueName}
+  tag <- AMQP.consumeMsgs consumeChannel queueName AMQP.NoAck $
+    replyCallback callback produceChannel
```

Comment:
hmm, but it isn't a response. here we are dealing only with replying.

or do you think `replyCallback` should create a response object and actual publish should happen here?